vllm/v1/worker/tpu_model_runner.py

```diff
@@ -1148,7 +1178,10 @@ def _precompile_sample_from_logits(self) -> None:
                         generate_params_if_all_greedy,
                     ))
                 sampling_metadata.all_greedy = all_greedy
-                self.sample_from_logits(dummy_logits, sampling_metadata)
+                with self.maybe_select_dummy_loras(
+                        self.lora_config, np.array([num_reqs],
+                                                   dtype=np.int32)):
```

Comment:
do we need to re-create this `np.array([num_reqs], dtype=np.int32))` at each inference step?
How about we only create it if we're actually using lora inside `maybe_select_dummy_loras`?

Reply:
It is only for compilation.